File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public int KthSmallest(TreeNode root, int k) {

        int res = root.val;
        int counter = 0;


        void dfs(TreeNode node){
            if(node is null) return;

            dfs(node.left);

            counter++;

            if(counter == k){
                res = node.val;
            }
            
            dfs(node.right);


        }

        dfs(root);

        return res;
        
    }
}

File: Data Structures & Algorithms/same-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} p
     * @param {TreeNode} q
     * @return {boolean}
     */
    isSameTree(p, q) {

        return this.traverse(p,"") == this.traverse(q,"")

    }

    traverse(root,code){
        if(!root) return "-"

        let newCode = code + root.val

        return newCode + this.traverse(root.left, newCode) + this.traverse(root.right, newCode)
    }
}
